### PR TITLE
EZP-25972: Clustering guide improvement

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-25972
| Versions      | all

Reorganizes sections as suggested in ticket description.

### CODE REVIEW

File: docs/guide/clustering.md [hunk 1]
```diff
@@ -15,13 +15,47 @@ is up to you and your performance needs.
...
+### Shared persistence cache
+
+[Memcached, a distributed caching solution](http://memcached.org/) is the main supported cache solution for clustering.
+
+See [Memcached Cache Adapter in Symfony documentation](https://symfony.com/doc/3.4/components/cache/adapters/memcached_adapter.html#configure-the-connection)
+for information on how to configure Memcached.
+
+!!! note
+
+    To use Memcached, you need to set `cache_service_name` to `cache.memcached`.
```
Comment (MagdalenaZuba) [inline docs/guide/clustering.md]: For me it's not an additional information but a step that must be taken so I would move it into the core text.

File: docs/guide/clustering.md [hunk 2]
```diff
@@ -35,12 +69,16 @@ For further details on requirements, see [Requirements page](../getting_started/
...
 
-It is also recommended to use:
+### Shared sessions
 
-- [Solr](solr.md) for better search and performance
-- a CDN for improved performance and faster ping time worldwide
-- active/passive database for failover
-- more recent versions of PHP and MySQL/MariaDB within [what is supported](../getting_started/requirements.md) for your eZ Platform version to get more performance out of each server. Numbers might vary so make sure to test this when upgrading.
+For a [cluster](clustering.md) setup you need to configure sessions to use a back end that is shared between web servers and supports locking.
+The only options out of the box supporting this in Symfony are the native PHP Memcached session save handler
```
Comment (MagdalenaZuba) [inline docs/guide/clustering.md]: The only out of the box options supporting

File: docs/guide/clustering.md [hunk 3]
```diff
@@ -35,12 +69,16 @@ For further details on requirements, see [Requirements page](../getting_started/
...
+provided by the `php-memcached` extension, and Symfony session handler for PDO (database).
+
+### Shared binary files
+
+eZ Platform supports multi-server setups by means of [custom IO handlers](file_management#the-dfs-cluster-handler).
+They make sure that files are correctly synchronized among the multiple clients that might use the data.
```
Comment (MagdalenaZuba) [inline docs/guide/clustering.md]: among the multiple clients using data.

File: docs/guide/clustering.md [hunk 4]
```diff
@@ -35,12 +67,16 @@ For further details on requirements, see [Requirements page](../getting_started/
...
     > Listen on &lt;addr&gt;; default to INADDR\_ANY. &lt;addr&gt; may be specified as host:port. If you don't specify a port number, the value you specified with -p or -U is used. You may specify multiple addresses separated by comma or by using -l multiple times. This is an important option to consider as there is no other way to secure the installation. Binding to an internal or firewalled network interface is suggested.
 
-It is also recommended to use:
+### Shared sessions
 
-- [Solr](solr.md) for better search and performance
-- a CDN for improved performance and faster ping time worldwide
-- active/passive database for failover
-- more recent versions of PHP and MySQL/MariaDB within [what is supported](../getting_started/requirements.md) for your eZ Platform version to get more performance out of each server. Numbers might vary so make sure to test this when upgrading.
+For a [cluster](clustering.md) setup you need to configure sessions to use a back end that is shared between web servers and supports locking.
```
Comment (andrerom) [inline docs/guide/clustering.md]: `.. , alternatively usage of a load balancer that supports session affinity ("sticky sessions") so a user is served by single server once he gets a session.`

Comment (andrerom) [reply docs/guide/clustering.md]: https://en.wikipedia.org/wiki/Load_balancing_(computing)#Persistence

**Comment by andrerom** [comment]
Yes, starting with 2.x we recommend Redis as performance on memcache is terrible _(In one extreme case I saw 0,5s vs 22s to load frontpage)_.

**Comment by andrerom** [comment]
Could we put this on hold until #418 is in? They overlap each other quite a bit.

**Comment by DominikaK** [comment]
Closing in favor of https://github.com/ezsystems/developer-documentation/pull/443 (too many conflicts)

**Pull request closed without merging**